xcat-inventory/xcclient/inventory/shell.py

```diff
@@ -31,35 +32,29 @@ class InventoryShell(shell.ClusterShell):
     @shell.arg('--dry', dest='dryrun', action="store_true", default=False, help='Dry run mode, nothing will be commited to xcat database')
     @shell.arg('-c','--clean', dest='update', action="store_false", default=True, help='clean mode. IF specified, all objects other than the ones to import will be removed')
     def do_import(self, args):
         """Import inventory file to xcat database"""
         mgr.validate_args(args, 'import')
-        if args.type :
-            #do export by type
-            mgr.import_by_type(args.type, args.name, args.path, dryrun=args.dryrun,version=args.version,update=args.update)
-        else :
-            mgr.import_all(args.path, dryrun=args.dryrun,version=args.version,update=args.update)
+        mgr.importobj(args.path,args.directory,args.type,args.name,dryrun=args.dryrun,version=args.version,update=args.update)
 
-    @shell.arg('-t','--type', metavar='<type>', help='type of objects to export, valid values: '+','.join(mgr.InventoryFactory.getvalidobjtypes())+'. '+'If not specified, all objects in xcat databse will be exported')
+    @shell.arg('-t','--type', metavar='<type>', help='comma "," delimited types of objects to export, valid values: '+','.join(mgr.InventoryFactory.getvalidobjtypes())+'. '+'If not specified, all objects in xcat databse will be exported')
     @shell.arg('-x', '--exclude', dest='exclude', default='', help='types to be excluded when exporting all, delimited with Comma(,).')
     @shell.arg('-o','--objects', dest='name',metavar='<name>', help='names of the objects to export, delimited with Comma(,). If not specified, all objects of the specified type will be exported')
-    @shell.arg('-f','--path', metavar='<path>', help='path of the inventory file(not implemented yet)')
+    @shell.arg('-f','--path', metavar='<path>', help='path of the inventory file')
+    @shell.arg('-d','--dir', dest='directory',metavar='<directory>', help='path of the inventory directory')
     @shell.arg('-s','--schema-version', dest='version',metavar='<version>', help='schema version of the inventory data. Valid values: '+','.join(mgr.InventoryFactory.getAvailableSchemaVersions())+'. '+'If not specified, the "latest" schema version will be used')
     @shell.arg('--format', metavar='<format>', help='format of the inventory data, valid values: json, yaml. json will be used by default if not specified ')
     def do_export(self, args):
         """Export the inventory data from xcat database"""
         mgr.validate_args(args, 'export')
-        if args.type :
-            # do export by type
-            mgr.export_by_type(args.type, args.name, args.path, args.format, version=args.version)
-        else :
-            mgr.export_all(args.path, args.format, exclude=args.exclude.split(','), version=args.version)
+        mgr.export_by_type(args.type, args.name, args.path,args.directory,args.format, version=args.version,exclude=args.exclude.split(','))
```

Comment:
please keep the coding format, we need a space after ','

Reply:
updated